### PR TITLE
Add 'linear' attribute to StationaryModel and InstationaryModel

### PR DESCRIPTION
**Pull request by pmli**
(Part of the changes from #652)

Since `rhs` and `mass` are asserted to be linear, it should be enough to add `self.linear = operator.linear`.

### CODE REVIEW

**Comment by pmli** [comment]
Right, for the input-output system to be linear, the output should also be linear (although a linear input-output system with a nonlinear operator and output could be constructed...).

File: src/pymor/models/basic.py [hunk 1]
```diff
@@ -126,6 +125,7 @@ def __init__(self, operator, rhs, outputs=None, products=None,
         self.rhs = rhs
         self.outputs = FrozenDict(outputs or {})
         self.solution_space = self.operator.source
+        self.linear = operator.linear and all(output.linear for output in outputs)
```
Comment (sdrave) [inline src/pymor/models/basic.py]: `outputs` is a dict so should be

`all(output.linear for output in outputs.values())`

File: src/pymor/models/basic.py [hunk 2]
```diff
@@ -249,6 +249,7 @@ def __init__(self, T, initial_data, operator, rhs, mass=None, time_stepper=None,
         self.time_stepper = time_stepper
         self.num_values = num_values
         self.outputs = FrozenDict(outputs or {})
+        self.linear = operator.linear and all(output.linear for output in outputs)
```
Comment (sdrave) [inline src/pymor/models/basic.py]: see above

File: src/pymor/models/basic.py (hunk 1 above)
Comment (pmli) [reply src/pymor/models/basic.py]: Of course. I've changed it to `self.outputs.values()` to not check for `outputs is None`.

**Pull request merged**